### PR TITLE
Docs: example about using azure storage account as storage

### LINKED ISSUE #2907: Add Azure blob storage config examples to the docs
**Is your feature request related to a problem? Please describe.**
Loki supports Azure Blob Storage, but this is not yet reflected in the docs. This issue is a reminder to add this.

### PR DESCRIPTION
**Pull request by charles-woshicai**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #2907 

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: docs/sources/storage/_index.md [hunk 1]
```diff
@@ -281,3 +285,33 @@ schema_config:
...
+    schema: v11
+    store: boltdb-shipper
+  storage_config:
+    azure:
+      # About how to get this value, see docs: https://docs.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage?tabs=azure-portal
+      account_key: xxx
+      # Your azure account name
+      account_name: xxx
+      # See https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction#containers
+      container_name: xxx
```
Comment (KMiller-Grafana) [inline docs/sources/storage/_index.md:308]: Please use a different substitution than xxx, as the other examples in this section have done.

```suggestion
    azure:
      # For the account-key, see docs: https://docs.microsoft.com/en-us/azure/storage/common/storage-account-keys-manage?tabs=azure-portal
      account_key: <account-key>
      # Your azure account name
      account_name: <account-name>
      # See https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction#containers
      container_name: <container-name>
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]
Please make suggested changes.  Then, this documentation addition will LGTM.

**Review by KMiller-Grafana — APPROVED** [review]
Thank you for making the changes. The docs LGTM now. Please take care of signing the CLA, such that we can merge these changes in.

**Comment by charles-woshicai** [comment]
> Thank you for making the changes. The docs LGTM now. Please take care of signing the CLA, such that we can merge these changes in.

Done.

**Review by owen-d — APPROVED** [review]
Thanks!

**Comment by hterik** [comment]
Hi
Being new to this, may i ask.

1. Why is request_timeout set to 0, when default value is 30.
It's easier to understand examples if they only contain the absolute minimum required configuration. If this deviation from the default is important, an explanation why would be good, and maybe in that case the default should be reconsidered.

2.  Why must the filesystem config be present?
It appears counter-inuitive that one has to add filesystem config when wanting to use azure.  Is this directory used for other temporary storage or is it a leftover from other config?

**Comment by thorker** [comment]
I stumpled at the same points.
Can someone please explain these configurations?

**Pull request merged**